heft-plugins/heft-webpack5-plugin/src/shared.ts

```diff
@@ -23,25 +23,26 @@ export type IWebpackConfiguration =
 /**
  * @public
  */
-export interface IWebpackBundleSubstageProperties extends IBundleSubstageProperties {
+ export interface IWebpack5PluginAccessor {
   /**
-   * The configuration used by the Webpack plugin. This must be populated
-   * for Webpack to run. If webpackConfigFilePath is specified,
-   * this will be populated automatically with the exports of the
-   * config file referenced in that property.
+   * A hook that allows for modification of the  configuration used by the Webpack
+   * plugin. This must be populated for Webpack to run. If a webpack configuration is
+   * provided, this will be populated automatically with the exports of the config
+   * file.
    *
    * @remarks
-   * Tapable event handlers can return `null` instead of `undefined` to suppress
+   * Tapable event handlers can return `false` instead of `undefined` to suppress
    * other handlers from creating a configuration object.
    */
-  // We are inheriting this problem from Tapable's API
-  // eslint-disable-next-line @rushstack/no-new-null
-  webpackConfiguration?: webpack.Configuration | webpack.Configuration[] | null;
-}
-
-/**
- * @public
- */
-export interface IWebpackBuildStageProperties extends IBuildStageProperties {
-  webpackStats?: webpack.Stats | webpack.MultiStats;
+  readonly onConfigureWebpackHook: AsyncSeriesWaterfallHook<IWebpackConfiguration | false>;
```

Comment:
In the past, we've split hooks and general data into two properties: `hooks` and `properties`. Are we keeping that convention around?

Sticking hooks inside a `hooks` property makes it obvious what they are without having to suffix their names with `Hook`.

Reply:
Agree, nicer having it on a property and trimming the suffix.